cli/main.rs

```diff
@@ -320,73 +316,80 @@ pub fn main() {
   );
 
   let args: Vec<_> = env::args_os().collect();
-
-  // NOTE(lucacasonato): due to new PKU feature introduced in V8 11.6 we need to
-  // initialize the V8 platform on a parent thread of all threads that will spawn
-  // V8 isolates.
-
   let current_exe_path = current_exe().unwrap();
   let standalone =
-    standalone::extract_standalone(&current_exe_path, args.clone());
+    standalone::extract_standalone(&current_exe_path, Cow::Borrowed(&args));
+
   let future = async move {
-    let standalone_res = match standalone {
+    match standalone {
       Ok(Some(future)) => {
         let (metadata, eszip) = future.await?;
         standalone::run(eszip, metadata).await
       }
-      Ok(None) => Ok(()),
-      Err(err) => Err(err),
-    };
-    // TODO(bartlomieju): doesn't handle exit code set by the runtime properly
-    unwrap_or_exit(standalone_res);
-
-    let flags = match flags_from_vec(args) {
-      Ok(flags) => flags,
-      Err(err @ clap::Error { .. })
-        if err.kind() == clap::error::ErrorKind::DisplayHelp
-          || err.kind() == clap::error::ErrorKind::DisplayVersion =>
-      {
-        err.print().unwrap();
-        std::process::exit(0);
+      Ok(None) => {
+        // NOTE(lucacasonato): due to new PKU feature introduced in V8 11.6 we need to
+        // initialize the V8 platform on a parent thread of all threads that will spawn
+        // V8 isolates.
+        let flags = resolve_flags_and_init(args)?;
+        run_subcommand(flags).await
```

Comment:
This feels even more confusing to me - maybe instead of doing this `Ok(Option)` and `Err(_)` matching we could return a more descriptive enum from `extract_standalone`? Something like:
```
enum StandaloneExtract {
  Found(future),
  NotFound,
  Error
}
```

Reply:
That match was there before so maybe it's just confusing because of the diff? I actually tried to type that return type to split up some of the functiosn more, but it's difficult because it's an `impl` return type and using a concrete type would be really verbose. I just pushed a commit that splits up the result handling from the option handling so it should be a bit clearer.